cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/PullOperationRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Request type for {@link PullOperation}.
+ */
+public class PullOperationRequest {
```

Comment:
should name it BulkPullRequest or something like that -- want to differentiate between bulk and single operations

Reply:
The plan is we would migrate the single pull to also use the same operation so BulkPull might be misleading.